src/Sarif/Readers/JsonPositionedTextReader.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Readers
+{
+    /// <summary>
+    ///  JsonPositionedTextReader is a JsonTextReader which exposes TokenPosition(),
+    ///  the byte offset in the stream where the current token begins. It is used to
+    ///  enable deferred collections, which look like other collections but are deserialized
+    ///  only when enumerated, saving memory.
+    /// </summary>
+    public class JsonPositionedTextReader : JsonTextReader
+    {
+        private LineMappingStreamReader StreamReader { get; set; }
+        private int LastReadLineNumber { get; set; }
```

Comment:
>private int LastReadLineNumber [](start = 8, length = 30)

since these are private, why not just use instance fields? just curious

Reply:
Made consistent with other classes and removed the two line numbers, which are unused and were leftover.